contracts/contracts/gateway/GatewayMessengerFacet.sol

```diff
@@ -62,6 +66,18 @@ contract GatewayMessengerFacet is GatewayActorModifiers {
             nonce: 0 // nonce will be updated by LibGateway.commitCrossMessage
         });
 
+        CrossMessageValidationOutcome outcome = LibGateway.validateCrossMessage(committed);
```

Comment:
why not just revert inside `validateCrossMessage`? Then there is no need to handle it outside.

Reply:
@karlem does it makes sense to have a helper function here for syntactic sugar? I'd keep the logic in `LibGateway#validateCrossMessage`, but would add a helper in `CrossMsgHelper` [bound to the `CrossMsg` type](https://solang.readthedocs.io/en/latest/language/using.html#binding-methods-to-types-with-using) so we can do:

```solidity
CrossMessageValidationOutcome outcome = committed.validateCrossMessage();
```

Reply:
@raulk yeah good idea.